index.js

```diff
@@ -1,9 +1,23 @@
 (function() {
-
   var RATE_LIMIT = 25;
 
-  var ComponentVisibilityMixin = {
+  /* Finding browser event names for hidden window event */
+  var hidden, visibilityChange;
+  if (typeof document.hidden !== "undefined") { // Opera 12.10 and Firefox 18 and later support
```

Comment:
good idea to rewrite this as a prefix array forEach instead.
```
['','moz','ms','webkit'].forEach(function(prefix) {
  if (typeof document[prefix + "hidden"] !== undefined) {
    hidden = prefix + "hidden";
    visibilityChange = prefix + "visibilitychange";
  }
});
```
That said, the target here is modern HTML5 browsers. If the latest versions of Firefox and Chrome, and IE 10 and 11, support this without prefixes, let's take out the prefixes.
```